File: Easy/Binary Search/Binary Search.cpp
#include <vector>
using namespace std;

int binarySearch(vector<int> array, int target)
{
    int left = 0, right = array.size() - 1, mid;
    while (left <= right)
    {
        mid = (left + right) / 2;
        if (array[mid] == target)
            return mid;
        else if (target < array[mid])
            right = mid - 1;
        else
            left = mid + 1;
    }
    return -1;
}

File: Easy/Binary Search/Binary Search.py
def binarySearch(array, target):
    left = 0
    right = len(array) - 1
    while left <= right:
        mid = (left + right)//2
        if array[mid] == target:
            return mid
        elif array[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
    return -1
